Makefile

```diff
@@ -3,7 +3,7 @@ SDIR=src
 TDIR=pytests
 # compilers
 TSC=./node_modules/typescript/bin/tsc
-TSCFLAGS=-t ES3 --sourceMap --module commonjs --noImplicitAny
```

Comment:
You should need `noImplicitAny` in there, as it catches a lot of bugs. Essentially, this tells the compiler to *never* assume that a value in the program is the `any` type, unless you explicitly give it that type. If the compiler is "fighting" you, you should look carefully at your program to determine why something is getting the `any` type, and add type annotations where needed.